backend/controllers/follows.js

```diff
@@ -1,13 +1,45 @@
 const Follows = require("./../models/Follows");
+const User = require("./../models/User");
 
 module.exports.createFollows = async (req, res) => {
   try {
+
     const followingUserId = req.params.followingUserId;
     const userId = req.user.id;
-    const follows = await Follows.create({ userId, followingUserId });
+
+    if (userId == followingUserId) {
+      return res.status(400).json({
+        message: "Users cannot folow themselves",
+      });
+    }
+
+    const [follows, created] = await Follows.findOrCreate({
+      where: { userId, followingUserId },
+      defaults: { userId, followingUserId }
+    });
+
+    if (!created) {
+      return res.status(200).json({
+        message: "The users are already following each other"
+      });
+    }
+
+    //increment following count for user
```

Comment:
Why do we need to fetch and then increment?
This adds an extra query.

Reply:
This is an older code, which would be removed after rebase